### PR TITLE
feature(routing): Removes /export and all secondary front controllers

### LINKED ISSUE #5017: Rethink how we do exports (Trac #5017)
_Original ticket http://trac.elgg.org/ticket/5017 on 43084402-06-08 by cash, assigned to unknown._

Elgg version: 1.8

Right now a lot of information is available if someone wants to make arbitrary requests on the export endpoint. We need to develop some use cases for export.

If exporting is primarily about giving users control over there data, only that user should be able to export the data. If it is about supporting migrating data to a new site/platform, doing it through http calls is tremendously inefficient.

### LINKED ISSUE #8425: Route all requests to index.php
Current system requires separate config in .htaccess, handler scripts, and blocks of code in Application::runCliServer, with no performance benefit. When design is right we basically shouldn't need ::runCliServer

### PR DESCRIPTION
**Pull request by mrclay**
This removes the “export” URL endpoints and removes all other /engine/handlers. Requests for these resources now all use the front controller `index.php`.

Fixes #8425
Fixes #5017

### CODE REVIEW

File: engine/classes/Elgg/Application.php [hunk 1]
```diff
@@ -269,76 +276,44 @@ public function getDb() {
 	}
 
 	/**
-	 * Rewrite rules for PHP cli webserver used for testing. Do not use on production sites
```
Comment (mrclay) [inline engine/classes/Elgg/Application.php]: Diff is messy but basically the separate runPhpWebServer() is going away.

File: engine/classes/Elgg/Application.php [hunk 2]
```diff
@@ -269,76 +276,44 @@ public function getDb() {
...
-		if (preg_match('/^\/export\/([A-Za-z]+)\/([0-9]+)\/([A-Za-z]+)\/([A-Za-z0-9\_]+)\/$/', $urlPath, $matches)) {
-			$_GET['view'] = $matches[1];
-			$_GET['guid'] = $matches[2];
-			$_GET['type'] = $matches[3];
-			$_GET['idname'] = $matches[4];
-			require "{$this->engine_dir}/handlers/export_handler.php";
-			exit;
+		if (php_sapi_name() === 'cli-server') {
+			$www_root = "http://{$_SERVER['SERVER_NAME']}:{$_SERVER['SERVER_PORT']}/";
+			$this->config->set('wwwroot', $www_root);
```
Comment (mrclay) [inline engine/classes/Elgg/Application.php]: FYI this is for #8426 (coming soon)

**Comment by ewinslow** [comment]
Love it. Wow what a breath of fresh air to get rid of all that extra cruft... Makes elgg/starter-project easier to support too.

LGTM

**Comment by mrclay** [comment]
I missed using a constant.

**Comment by mrclay** [comment]
Updated

**Comment by ewinslow** [comment]
The constants are nice too.

**Pull request merged**